File: SchoolWebApp-API/Project.Core/Entities/CBE/Assessments/Strand.cs
using SchoolWebApp.Core.Entities.Academics;
using SchoolWebApp.Core.Entities.Class;
using SchoolWebApp.Core.Entities.Shared;

namespace SchoolWebApp.Core.Entities.CBE.Assessments
{
    public class Strand : SettingsBase
    {
        public string? Code { get; set; }

        public int? CurriculumId { get; set; }
        public Curriculum? Curriculum { get; set; } = null;

        public int Version { get; set; }

        public int SubjectId { get; set; }
        public Subject? Subject { get; set; } = null;

        public int LearningLevelId { get; set; }
        public LearningLevel? LearningLevel { get; set; } = null;

        public int? ThemeId { get; set; }
        public Theme? Theme { get; set; } = null;

        public List<SubStrand> SubStrands { get; set; } = new();
    }
}

File: SchoolWebApp-Client/src/app/cbe/assessments/models/strand.ts
import { Curriculum } from '@/academics/models/curriculum';
import { Subject } from '@/academics/models/subject';
import { LearningLevel } from '@/class/models/learning-level';
import {ResourceModel} from '@/core/models/ResourceModel';
import { AcademicYear } from '@/school/models/academic-year';
export class Strand extends ResourceModel<Strand> {
    public name?: string;
    public code?: string;
    public description?: string
    public rank?: number;

    public curriculumId?: number;
    public curriculum?: Curriculum;

    public learningLevelId?: number;
    public learningLevel?: LearningLevel;
    
    public academicYearId?: number;
    public academicYear?: AcademicYear;

    public subjectId?: number;
    public subject?: Subject;

    public themeId?: number;
    public theme?: any;

    constructor(model?: Partial<Strand>) {
        super(model);
    }
}
